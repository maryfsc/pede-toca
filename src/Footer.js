import React from 'react';
import { Row } from 'reactstrap';

const Footer = () => {
  return (
    <footer className="container-fluid">
      <Row>
        <div className="col-lg-12">
          <p className="footer-message text-white text-center">A cool site made with React & PedeToca API!</p>
        </div>
      </Row>
    </footer>
    )
}

export default Footer;