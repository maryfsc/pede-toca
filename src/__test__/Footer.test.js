import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../Footer';

describe('<Footer />', () => {
  const wrapper = shallow(<Footer />);

  it('renders correctly', () => {
    expect(wrapper.find(<footer id="footer"></footer>)).toBeDefined();
    expect(wrapper.find(<div className="footer-message"></div>)).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});



// nesse caso não estou testando exatamente o texto contido no footer (children), né? só o snapshot do componente :thinking: 