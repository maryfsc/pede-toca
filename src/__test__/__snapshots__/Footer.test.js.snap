// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row
        tag="div"
      >
        <div
          className="col-lg-12"
        >
          <p
            className="footer-message text-white text-center"
          />
        </div>
      </Row>,
      "className": "container-fluid",
      "id": "footer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="col-lg-12"
        >
          <p
            className="footer-message text-white text-center"
          />
        </div>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="footer-message text-white text-center"
          />,
          "className": "col-lg-12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "footer-message text-white text-center",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        "type": "div",
      },
      "type": [Function],
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row
          tag="div"
        >
          <div
            className="col-lg-12"
          >
            <p
              className="footer-message text-white text-center"
            />
          </div>
        </Row>,
        "className": "container-fluid",
        "id": "footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="col-lg-12"
          >
            <p
              className="footer-message text-white text-center"
            />
          </div>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="footer-message text-white text-center"
            />,
            "className": "col-lg-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "footer-message text-white text-center",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
